test/protected_resource_test.rb

```diff
@@ -0,0 +1,101 @@
+require 'test_helper'
+
+class TestProtectedResource < Minitest::Test
+  include CircuitBreakerHelper
+  include ResourceHelper
+
+  def setup
+    Semian.destroy(:testing)
+  rescue
+    nil
+  end
+
+  def teardown
+    destroy_resources
+  end
+
+  def test_acquire_without_bulkhead
+    Semian.register(
+      :testing,
+      exceptions: [SomeError],
+      error_threshold: 2,
+      error_timeout: 5,
+      success_threshold: 1,
+      bulkhead: false,
+    )
+
+    10.times do
+      Thread.new do
+        block_called = false
+        @resource = Semian[:testing]
+        @resource.acquire { block_called = true }
+        assert_equal true, block_called
+        assert_instance_of Semian::CircuitBreaker, @resource.circuit_breaker
+        assert_nil @resource.bulkhead
+      end.join
+    end
+  end
+
+  def test_register_with_circuit_breaker_missing_options
+    assert_raises ArgumentError do
+      Semian.register(
+        :testing,
+        error_threshold: 2,
+        error_timeout: 5,
+        bulkhead: false,
+      )
+    end
+  end
+
+  def test_register_with_bulkhead_missing_options
+    assert_raises ArgumentError do
+      Semian.register(
+        :testing,
+        circuit_breaker: false,
+      )
+    end
+  end
+
+  def test_acquire_bulkhead_without_circuit_breaker
```

Comment:
This doesn't test that the circuit breaker is not present other than checking the value. I don't really have a strong opinion on these tests, but they seem to have a bit of an identity crisis between checking what's enabled and integration testing. I think I suggested the latter, but that would assume you could use some helpers or something from the implementation tests.